tools/vms-generator/vms-generator.go

```diff
@@ -187,9 +190,39 @@ func main() {
 		return nil
 	}
 
+	admitter := validating_webhook.VMsAdmitter{
+		InstancetypeMethods: &testutils.MockInstancetypeMethods{
+			FindInstancetypeSpecFunc: func(vm *v1.VirtualMachine) (*instancetypev1beta1.VirtualMachineInstancetypeSpec, error) {
+				if vm.Spec.Instancetype != nil && vm.Spec.Instancetype.Name != "" {
+					kind := strings.ToLower(vm.Spec.Instancetype.Kind)
+					name := vm.Spec.Instancetype.Name
+					if kind == "" || kind == apiinstancetype.ClusterSingularResourceName {
+						if vmClusterInstancetype, ok := virtualMachineClusterInstancetypes[name]; ok {
+							return &vmClusterInstancetype.Spec, nil
+						}
+					} else if kind == apiinstancetype.SingularResourceName {
+						if vmInstancetype, ok := virtualMachineInstancetypes[name]; ok {
+							return &vmInstancetype.Spec, nil
+						}
+					}
+				}
+				return nil, nil
+			},
+			FindPreferenceSpecFunc: func(vm *v1.VirtualMachine) (*instancetypev1beta1.VirtualMachinePreferenceSpec, error) {
+				return nil, nil
+			},
+			ApplyToVmiFunc: func(field *k8sfield.Path, instancetypespec *instancetypev1beta1.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *v1.VirtualMachineInstanceSpec, vmiMetadata *metav1.ObjectMeta) instancetype.Conflicts {
```

Comment:
Can you not just point to the actual function?

`ApplyToVmiFunc: instancetypeMethods.ApplyToVmi,`

Reply:
No, it is a method on the `InstancetypeMethods` struct.

Reply:
Right but you can define `instancetypeMethods` outside of this and just pass the method still right?

```
instancetypeMethods := instancetype.InstancetypeMethods{}
admitter := validating_webhook.VMsAdmitter{
    InstancetypeMethods: &testutils.MockInstancetypeMethods{
        [..]
        ApplyToVmiFunc: instancetypeMethods.ApplyToVmi,

```

This works for me locally at least. Just a nit really so feel free to ignore.

Reply:
Ah, got it, of course!